plugins/obs-webrtc/whip-output.cpp

```diff
@@ -134,24 +140,70 @@ void WHIPOutput::ConfigureVideoTrack(std::string media_stream_id,
 		media_stream_track_id.c_str(),
 	};
 
+	// generate the random starting ts for the video track
+	uint32_t rtp_video_timestamp = generate_random_u32();
+
 	rtcPacketizationHandlerInit packetizer_init = {
 		video_ssrc,
 		cname.c_str(),
 		video_payload_type,
 		video_clockrate,
 		0,
-		0,
+		rtp_video_timestamp,
 		RTC_NAL_SEPARATOR_START_SEQUENCE,
-		0};
+		max_fragment_size};
 
 	video_track = rtcAddTrackEx(peer_connection, &track_init);
 	rtcSetH264PacketizationHandler(video_track, &packetizer_init);
 	rtcChainRtcpSrReporter(video_track);
 	rtcChainRtcpNackResponder(video_track, 1000);
 }
 
-bool WHIPOutput::Setup()
+/**
+ * @brief Init before OPTIONS and Setup due to the need of the endpoint url for SendOptions.
+ * 
+ */
+bool WHIPOutput::Init()
 {
+	// construct operating system info string
+	std::stringstream os_info;
```

Comment:
Do we need to use `std::stringstream` instead of just `const char *` or `std::string`?

Reply:
I cannot find such a usage that the operator `<<` is used only once for a `std::stringstream` variable.

Reply:
@norihiro right, but let's not forget your mod which I pulled in :)  I'll get it changed